dealership/contracts.py

```diff
@@ -4,9 +4,29 @@ class Contract(object):
 
 class BuyContract(Contract):
     def __init__(self, vehicle, customer, monthly_payments):
-        pass
+        self.vehicle = vehicle
+        self.customer = customer
+        self.monthly_payments = monthly_payments
+    
+    def total_value(self):
+        interest = self.vehicle.interest_rate * self.monthly_payments * self.vehicle.sale_price() / 100
+        discount_multiplier = 1 - (self.customer.discount if self.customer.is_employee() else 0)
+        return discount_multiplier * (self.vehicle.sale_price() + interest)
+    
+    def monthly_value(self):
+        return self.total_value() / self.monthly_payments
 
 
 class LeaseContract(Contract):
     def __init__(self, vehicle, customer, length_in_months):
-        pass
+        self.vehicle = vehicle
+        self.customer = customer
+        self.length_in_months = length_in_months
+
+    def total_value(self):
+        lease_cost = (self.vehicle.sale_price() * self.vehicle.lease_multiplier) / self.length_in_months
+        discount_multiplier = 1 - (self.customer.discount if self.customer.is_employee() else 0)
+        return discount_multiplier * (self.vehicle.sale_price() + lease_cost)
```

Comment:
A few of these lines are repeated with the ones in `BuyContract`. You could refactor it to have the common code sitting in the parent class (`Contract`) and then each subclass implementing its own particular functionality. Check our solution: https://github.com/rmotr-group-projects/pyp-u1-c1-car-dealership/blob/solution/dealership/contracts.py